### PR TITLE
Rewind: add flow in AL to switch from VP/activate Rewind/migrate credentials

### LINKED ISSUE #21850: Vaultpress to Jetpack transition flow
The latest screenshots and flows can be found in the prototype. I **strongly** suggest familiarizing yourself with the prototype for full flows: p9rlnk-34

For ease of implementation, I've broken this down into unique screens, implementation details, and their assets. The first list is for tracking implementation. The details for each are just past the list.

- [ ] 1. JITM Banner
- [x] 2. Backups and security welcome
- [x] 3a. Migrate credentials prompt
- [x] 3b. Add your credentials
- [x] 4. Credentials form
- [x] 5. Success message

## 1. JITM Banner

These banners will be the entry points into the flow from Jetpack for the following user types:
* Users with Jetpack plan with or without Vaultpress with or without credentials
* Users with Jetpack without a plan (free)

**These will not be shown for anyone with a Vaultpress plan. (At least not yet)**

vp-creds jetpack dashboard banner
vp-creds jetpack settings banner
vp-creds wp-admin dashboard banner

### Copy
**Headline**
> Try Jetpack's powerful new backup and security

**Subhead**
> With a Jetpack plan, you can rewind your site to any event in its history

**Button - folks with Vaultpress**
> Switch now

**Button - folks without Vaultpress**
> Set up

**Button - folks without a plan**
> Upgrade

### Implementation notes
* The button text will change depending on user flow
* The banner will appear in the wp-admin Dashboard, the Jetpack Dashboard, and the Jetpack Settings > Security > Backups and security scanning group

## 2. Backups and security welcome
This will be the next step for folks who have clicked through the variety of JITM banners (including one on the Activity Log page). They will likely need to log in before they can see this.

vp-no-creds welcome

### Copy
**Headline**
> Welcome to Jetpack's new backups and security

**SVG**
[security.svg](https://github.com/Automattic/wp-calypso/blob/master/public/images/illustrations/security.svg)

> Backing up and securing your site should be a breeze. Our new seamless system makes it possible to see all your site's activity from one convenient dashboard.

**Button - folks with Vaultpress**
> Switch now

**Button - folks without Vaultpress**
> Continue setup

**Button - folks without a plan**
> Upgrade now

**Cancel link**
> No thanks

**Subhead**
> What else can it do?

**Section 1 heading**
> Rewind to any event

**Section 1 copy**
> As soon as you switch over, we will start tracking every change made to your site and allow you to rewind to any past event. If you lose a file, get hacked, or just liked your site better before some changes, you can rewind with a click of a button.

**Section 1 SVG**
[backup.svg](https://github.com/Automattic/wp-calypso/blob/master/public/images/illustrations/backup.svg)

**Section 2 heading**
> Stay on top of your site's security

**Section 2 copy**
> When something happens to your website you want to know it immediately. And you will – we'll send you instant alerts based on our 24/7 monitoring of advanced threats.

**Section 2 SVG**
[security-issue.svg](https://cldup.com/wlmSPmuUKU.svg) (might need to be added to the Calypso `images/illustrations/` directory)

**Section 3 heading**
> Log all events on your site

**Section 3 copy**
> Access a new, streamlined history of events on your site – from published posts to user-role changes. If you ever need to figure out what happened when, now you can get the answer in seconds.

**Section 3 SVG**
[stats.svg](https://cldup.com/T4IGTV0IKa.svg) (needs to be added to the Calypso `images/illustrations/` directory)

### Implementation notes
* The button text will change depending on user flow
* When `No thanks` is selected, it just takes them back to where they were previously in wp-admin or Calypso

## 3a. Migrate credentials prompt
This will only show for _users with Jetpack plan with Vaultpress with valid credentials_. There is another version for other user flows.

vp-creds migrate credentials

### Copy
**Headline**
> Migrate credentials

**SVG**
[almost-there.svg](https://cldup.com/xmouy_FNKy.svg) (needs to be added to the Calypso `images/illustrations/` directory)

**Copy**
> You've already configured VaultPress correctly, so we're ready to migrate your credentials over to Jetpack with just one click. Are you ready to switch to our faster, more powerful system? Let's go!

**Button**
> Migrate your credentials

**Note** (this should reuses similar styles if possible maybe from the login or someplace)
> Note: Moving to Jetpack backups and security is final, and the VaultPress backups previously generated will not be migrated to the new system. We will retain the data in case you need to restore to a backup made before you switched.

### Implementation notes
* Upon clicking `Migrate your credentials` the user is direct to the "We're backing up your site!" view
* There is a back button here. It should just go back to the previous view.

## 3b. Add your credentials
This will show for _users with Jetpack plan with or without Vaultpress without credentials_.

jp-free approve credentials

### Copy

**Headline**
> Add your credentials

**SVG**
[security.svg](https://github.com/Automattic/wp-calypso/blob/master/public/images/illustrations/security.svg)

**Copy**
> To switch you over to Jetpack backups and security we need you to add your site credentials. This gives WordPress.com access to your site to perform automatic actions on your server -- like backups. It also allows us to restore your site and manually access it in case of an emergency.

**Button**
> Add your credentials

## Implementation details
* Clicking `Add your credentials` opens the Credentials form
* There is a back button here. It should just go back to the previous view.

## 4. Credentials form
This form will use the credentials validation/success notices already in place so I won't show them here, but here's the latest frankenmockup:

jp-paid server credentials

## 5. Success message
If everything goes smoothly with credentials, this is the message that will appear. Huzzah! If not, hopefully the errors were caught before now on the credentials form view and worked out with a Happiness Engineer if necessary.

jp-paid approved credentials

## Copy
**Headline**
> We're backing up your site!

**SVG**
[thankYou.svg](https://github.com/Automattic/wp-calypso/blob/master/public/images/illustrations/thankYou.svg)

**Copy**
> Welcome to your Jetpack Premium plan! That sound you're hearing? It's your website doing backflips with excitement!

**Button - folks who just bought a plan**
> Return to your site

**Button - everyone else**
> View your activity

### Implementation notes

* The button copy will change only for folks who just purchased a plan. This is because they might not have much activity yet and will receive notifications/emails later to lead them to the activity log.
* There probably shouldn't be a back button. ¯\\_(ツ)_/¯

### PR DESCRIPTION
**Pull request by eliorivero**
Fixes #21850

Introduces a dialog that prompts user to switch from VP to Rewind or introduce credentials.

captura de pantalla 2018-01-26 a la s 19 33 14

### Dismiss

When users click the "No thanks" link, they're taken back to where they were in WP Admin. For this, the query parameter `rewind-redirect` is needed. It points to a page in WP Admin like this `rewind-redirect=admin.php?page=somepage`.

### Testing

When you arrive to AL with a URL like:
```
http://calypso.localhost:3000/stats/activity/sleepy-spectacled.jurassic.ninja?rewind-redirect=admin.php%3Fpage%3Djetpack
```
- verify you get the dialog
- verify that the "No thanks" link takes you back to `admin.php?page=whatever`

Depending on the site setup there are three particular flows:

a. Create a site with Jetpack and VaultPress plugins: it must have a Jetpack plan with backups and VaultPress be active. It must not have a VP plan. 
- verify you get the **Switch now** button in the dialog
- in the flow, you'll get a button to switch the site from VP to Rewind.

b. Create a site that has a Jetpack plan with backups but never had VP installed.
- verify you get the **Continue setup** button in the dialog
- in the flow, you'll get the form to add the credentials.

c. Create a site with a Jetpack free plan. Purchase a plan with credits.
- when you're in the orange **Thank You** screen, verify you have a **Setup backups** button
- click the button and verify you're put into the flow to add the credentials.

### CODE REVIEW

File: client/my-sites/stats/activity-log-switch/index.jsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+					/>
+				);
+
+			default:
+				return (
+					<Button
+						primary
+						href={ `/settings/security/${ siteSlug }` }
+						>
+						{ translate( 'Continue setup' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log-switch/index.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Continue your site setup:' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/stats/activity-log-switch/index.jsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			siteId,
+			siteSlug,
+			translate,
+		} = this.props;
+		switch ( this.props.failureReason ) {
+			case 'vp_can_transfer':
+				return (
+					<ActivityLogRewindToggle
+						siteId={ siteId }
+						label={ translate( 'Switch now' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log-switch/index.jsx]: Hi! I've found a possible matching string that has already been translated **8** times:
`translate( 'Switch' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/stats/activity-log-switch/index.jsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+			translate,
+		} = this.props;
+		switch ( this.props.failureReason ) {
+			case 'vp_can_transfer':
+				return (
+					<Button
+						primary
+						href={ `/start/rewind-switch/?siteId=${ siteId }&siteSlug=${ siteSlug }` }
+					>
+						{ translate( 'Switch now' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log-switch/index.jsx]: Hi! I've found a possible matching string that has already been translated **8** times:
`translate( 'Switch' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/stats/activity-log-switch/index.jsx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+				);
+
+			case 'awaitingCredentials':
+			default:
+				return (
+					<Button
+						primary
+						href={ `/start/rewind-setup/?siteId=${ siteId }&siteSlug=${ siteSlug }` }
+						>
+						{ translate( 'Continue setup' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log-switch/index.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Continue your site setup:' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/rewind-add-creds/index.jsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+		this.props.goToStep( 'rewind-form-creds' );
+	};
+
+	stepContent = () => {
+		const { translate } = this.props;
+
+		return (
+			<Card className="rewind-add-creds__card rewind-switch__card rewind-switch__content">
+				<h3 className="rewind-add-creds__title rewind-switch__heading">
+					{ translate( 'Add your credentials' ) }
```
Comment (a8ci18n) [inline client/signup/steps/rewind-add-creds/index.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Add credentials' )` *ES Score: 12*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/rewind-form-creds/index.jsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+		siteSlug: PropTypes.string.isRequired,
+	};
+
+	stepContent = () => {
+		const { translate, siteSlug } = this.props;
+
+		return (
+			<Card className="rewind-form-creds__card rewind-switch__card rewind-switch__content">
+				<h3 className="rewind-form-creds__title rewind-switch__heading">
+					{ translate( 'Site credentials' ) }
```
Comment (a8ci18n) [inline client/signup/steps/rewind-form-creds/index.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Add site credentials' )` *ES Score: 15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/signup/steps/rewind-migrate/index.jsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+	stepContent = () => {
+		const { translate, siteId } = this.props;
+
+		return (
+			<div className="rewind-migrate__card rewind-switch__card">
+				<Card className="rewind-migrate__content rewind-switch__content">
+					<h3 className="rewind-migrate__title rewind-switch__heading">
+						{ translate( 'Migrate credentials' ) }
+					</h3>
+					<img src="/calypso/images/illustrations/almost-there.svg" alt="" />
```
Comment (enejb) [inline client/signup/steps/rewind-migrate/index.jsx]: Should we add the alt text here?

Comment (eliorivero) [reply client/signup/steps/rewind-migrate/index.jsx]: Decorative images are not required to have a text. They can just have an empty alt attribute.

File: client/signup/steps/rewind-were-backing/index.jsx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+	stepContent = () => {
+		const { translate, siteSlug } = this.props;
+
+		return (
+			<Card className="rewind-were-backing__card rewind-switch__card rewind-switch__content">
+				<h3 className="rewind-were-backing__title rewind-switch__heading">
+					{ translate( "We're backing up your site!" ) }
+				</h3>
+				<img src="/calypso/images/illustrations/thankYou.svg" alt="" />
```
Comment (enejb) [inline client/signup/steps/rewind-were-backing/index.jsx]: Same alt tag question.

**Comment by enejb** [comment]
The code looks good. I noticed that all the alt tags are missing, lets add them. 
Also it would be good to add some sort of event tracking ( not sure if this taken care of though a different component) But it would be good to add that, probably in a different PR.

**Comment by MichaelArestad** [comment]
Visually, it's pretty good, but I have some suggestions.

The first, I suppose I didn't make as clear as I could have, is that we could probably reuse most of the styles from the WordPress.com thank page after a purchase:

image

The second suggestion is that we maybe should hide the activity on the page while this view is showing.

**Comment by enejb** [comment]
So the loading order of the page is a bit strange 
1. First I see the activity log actions. 
screen shot 2018-01-30 at 1 43 36 pm

2. Then I see the notice but it doesn't have the images. 
screen shot 2018-01-30 at 1 43 38 pm

3. Then the image show up. Can we make them show up at the same time (inline the svg )? 
screen shot 2018-01-30 at 1 43 25 pm


When I click Continue setup  I see 
screen shot 2018-01-30 at 1 45 26 pm

And then 
screen shot 2018-01-30 at 1 45 34 pm

I think there should be some space to see the step numbers as expected.

**Comment by enejb** [comment]
I think we should fix the load order a bit so that it show up nicer.

**Comment by eliorivero** [comment]
oh @enejb I forgot to mention that I need to _not_ render all the AL elements when this is rendered. I'll do that last, since I don't want to get into conflicts with things in AL that change frequently due to all our changes. I'll set this back to in progress and will ping you for review once it's final final. Thanks!

**Comment by eliorivero** [comment]
I pushed the change that completely hides AL when the flow initiator dialog is shown.

@enejb @MichaelArestad can we leave the SVG inlining and styles load for a separate PR? I'll handle it immediately after this is merged

**Comment by enejb** [comment]
Yes lets leave the svg as it is for now.

**Comment by enejb** [comment]
I just tested this and after going to though the credentials process I get stuck there. 

I noticed that the Step # are gone. So that part worked. Although I think it does help the user for them to be there.

I use a Jurassic site and followed the steps are updating to a plan.
It would be good to prepopulate the ABSPATH for the user. Since it is not very easy to find sometimes.

**Comment by samhotchkiss** [comment]
> It would be good to prepopulate the ABSPATH for the user. Since it is not very easy to find sometimes.

This isn't needed-- we shouldn't even show the path (it should be under "advanced")-- we auto-populate the abspath if the user doesn't manually set their own path.

**Comment by enejb** [comment]
I tried adding the credentials without the `ABSPATH` and I wasn't able to connect the site without entering one. 😿

**Comment by eliorivero** [comment]
@enej thanks for reporting the issue with creds form going nowhere. I've solved this issue in https://github.com/Automattic/wp-calypso/pull/21802/commits/00b38eb5abb567d3b2bdcc03697230521d126069

Regarding the ABSPATH, it's something that we discussed in calls. We're aiming to have some method to detect the WP directory but that has to happen in the back end. It's not the scope of this PR since this only reuses the creds form and we can tackle that later.

**Comment by enejb** [comment]
I tested the 3 different flows. 
Some things that I noticed. 
- (a) The migration flow worked well once the user was fully connected. ( I think we were missing some user permissions ) 
- (b) This flow worked well. Although I think the adding youre credentails could work better. 
They should also be some checking to make sure that we have the right site object. At one point I was trying to update the site_id 0. Which doesn't seem it should be possible. ( I think that should be addressed in a different PR) This one is big enough already. 
- Also I needed to specify the ABSPATH which I think should never be the case. ( As a side note VaultPress has a much nicer way of adding these credentials. It doesn't ask you for the ABSPATH and it gives clear instructions as to what you need to do with the public key that it auto-generates for you. ❤️ 
- (c) Flow c seems to be conflicting with what @rcoll is doing.

**Review by enejb — APPROVED** [review]
Lets merge this as and iterate over things.

**Comment by ddc22** [comment]
Is this flow something that we can cleanup?

https://wordpress.com/start/rewind-were-backing

Does not work.
Or maybe I'm testing it wrong?

**Pull request merged**